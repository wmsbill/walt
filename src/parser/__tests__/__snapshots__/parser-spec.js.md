# Snapshot report for `src/parser/__tests__/parser-spec.js`

The actual snapshot is saved in `parser-spec.js.snap`.

Generated by [AVA](https://ava.li).

## compiles exports

> Snapshot 1

    {
      Code: [],
      Element: [],
      Exports: [
        {
          field: 'answer',
          index: 0,
          kind: 3,
        },
      ],
      Functions: [],
      Globals: [
        {
          init: 42,
          mutable: 0,
          type: 127,
        },
      ],
      Imports: [],
      Memory: [],
      Types: [],
      body: [
        {
          Type: '',
          decl: {
            Type: 'Declaration',
            const: true,
            globalIndex: 0,
            id: 'answer',
            init: {
              Type: 'Constant',
              meta: [],
              params: [],
              range: [
                {
                  col: 27,
                  line: 1,
                },
                {
                  col: 29,
                  line: 1,
                },
              ],
              type: 'i32',
              value: '42',
            },
            meta: [],
            params: [],
            range: [
              {
                col: 7,
                line: 1,
              },
              {
                col: 30,
                line: 1,
              },
            ],
            type: 'i32',
            value: 'const',
          },
          meta: [],
          params: [],
          range: [
            {
              col: 0,
              line: 1,
            },
          ],
          value: 'export',
        },
      ],
    }

## compiles globals

> Snapshot 1

    {
      Code: [],
      Element: [],
      Exports: [],
      Functions: [],
      Globals: [
        {
          init: 42,
          mutable: 0,
          type: 127,
        },
      ],
      Imports: [],
      Memory: [],
      Types: [],
      body: [
        {
          Type: 'Declaration',
          const: true,
          globalIndex: 0,
          id: 'answer',
          init: {
            Type: 'Constant',
            meta: [],
            params: [],
            range: [
              {
                col: 20,
                line: 1,
              },
              {
                col: 22,
                line: 1,
              },
            ],
            type: 'i32',
            value: '42',
          },
          meta: [],
          params: [],
          range: [
            {
              col: 0,
              line: 1,
            },
            {
              col: 23,
              line: 1,
            },
          ],
          type: 'i32',
          value: 'const',
        },
      ],
    }

## the most basic of modules in wasm

> Snapshot 1

    {}
