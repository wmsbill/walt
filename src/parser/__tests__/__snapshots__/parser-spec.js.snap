// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles exports 1`] = `
Object {
  "Code": Array [],
  "Element": Array [],
  "Exports": Array [
    Object {
      "field": "answer",
      "index": 0,
      "kind": 3,
    },
  ],
  "Functions": Array [],
  "Globals": Array [
    Object {
      "init": 42,
      "mutable": 0,
      "type": 127,
    },
  ],
  "Imports": Array [],
  "Types": Array [],
  "body": Array [
    Object {
      "decl": Object {
        "Type": "Declaration",
        "const": true,
        "end": Object {
          "col": 30,
          "line": 1,
        },
        "globalIndex": 0,
        "id": "answer",
        "init": Object {
          "Type": "Constant",
          "end": Object {
            "col": 29,
            "line": 1,
          },
          "range": Array [
            Object {
              "col": 27,
              "line": 1,
            },
            Object {
              "col": 29,
              "line": 1,
            },
          ],
          "start": Object {
            "col": 27,
            "line": 1,
          },
          "value": "42",
        },
        "range": Array [
          Object {
            "col": 7,
            "line": 1,
          },
          Object {
            "col": 30,
            "line": 1,
          },
        ],
        "start": Object {
          "col": 7,
          "line": 1,
        },
        "type": "i32",
      },
      "range": Array [
        Object {
          "col": 0,
          "line": 1,
        },
      ],
      "start": Object {
        "col": 0,
        "line": 1,
      },
    },
  ],
}
`;

exports[`compiles globals 1`] = `
Object {
  "Code": Array [],
  "Element": Array [],
  "Exports": Array [],
  "Functions": Array [],
  "Globals": Array [
    Object {
      "init": 42,
      "mutable": 0,
      "type": 127,
    },
  ],
  "Imports": Array [],
  "Types": Array [],
  "body": Array [
    Object {
      "Type": "Declaration",
      "const": true,
      "end": Object {
        "col": 23,
        "line": 1,
      },
      "globalIndex": 0,
      "id": "answer",
      "init": Object {
        "Type": "Constant",
        "end": Object {
          "col": 22,
          "line": 1,
        },
        "range": Array [
          Object {
            "col": 20,
            "line": 1,
          },
          Object {
            "col": 22,
            "line": 1,
          },
        ],
        "start": Object {
          "col": 20,
          "line": 1,
        },
        "value": "42",
      },
      "range": Array [
        Object {
          "col": 0,
          "line": 1,
        },
        Object {
          "col": 23,
          "line": 1,
        },
      ],
      "start": Object {
        "col": 0,
        "line": 1,
      },
      "type": "i32",
    },
  ],
}
`;

exports[`the most basic of modules in wasm 1`] = `Object {}`;
